ScanToShop/ViewControllers/LoginViewController.m

```diff
@@ -19,12 +21,46 @@ @implementation LoginViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    self.passwordField.secureTextEntry = YES;
+}
+
+- (IBAction)onViewTap:(id)sender {
+    [self.view endEditing:YES];
 }
 
 - (IBAction)onSignInTap:(id)sender {
+    [self loginUser];
 }
 
 - (IBAction)onRegisterTap:(id)sender {
+    self.usernameField.text = @"";
+    self.passwordField.text = @"";
+}
+
+- (void)loginUser {
+    if ([self areUserInputFieldsEmpty]) {
+        [AlertManager loginAlert:@"" ViewController:self];
+        return;
+    }
+    
+    NSString *username = self.usernameField.text;
+    NSString *password = self.passwordField.text;
+    
+    [PFUser logInWithUsernameInBackground:username password:password block:^(PFUser * user, NSError *  error) {
+        if (error) {
+            NSLog(@"User log in failed: %@", error.localizedDescription);
+            [AlertManager loginAlert:error.localizedDescription ViewController:self];
+        } else {
+            NSLog(@"User logged in successfully");
+            [self performSegueWithIdentifier:@"loginSegue" sender:nil];
+        }
+    }];
 }
 
+-(BOOL) areUserInputFieldsEmpty {
```

Comment:
Like the helper method here.

Reply:
Lol. Not to torture you but what about new lines and white space? I won't mind if you save that for final touches after app is done. Just another thing to think about.